.travis.yml

```diff
@@ -0,0 +1,87 @@
+language: c
+branches:
+  only:
+  - master
+
+dist: xenial
+
+cache:
+  directories:
+    - $HOME/.cabsnap
+
+matrix:
+  include:
+    - env: TEST=MAIN GHC_VER=8.4.4 BUILD=CABAL CABAL_VER=2.2 STDLIB_VER=1.2
+      addons:
+        apt:
+          packages:
+            - cabal-install-2.2
+            - ghc-8.4.4
+          sources:
+            - hvr-ghc
+
+before_install:
+  - export PATH=/opt/ghc/$GHC_VER/bin:/opt/cabal/$CABAL_VER/bin:/opt/alex/3.1.7/bin:/opt/happy/1.19.5/bin:~/.cabal/bin/:$PATH;
```

Comment:
pinning down alex and happy indirectly like this also makes it seem problematic (for later maintenance).